doc/source/development/roadmap.rst

```diff
@@ -0,0 +1,153 @@
+.. _roadmap:
+
+=======
+Roadmap
+=======
+
+This page provides an overview of the major themes pandas' development. Implementation
+of these goals may be hastened with dedicated funding.
+
+Extensibility
+-------------
+
+Pandas :ref:`extending.extension-types` provide 3rd-party libraries the ability to
+extend pandas' supported data types. Many parts of pandas still unintentionally
+convert data to a NumPy array. These problems are especially pronounced for nested data.
+
+We'd like to improve the handling of extension arrays throughout the library,
+making their behavior more consistent with the handling of NumPy arrays. We'll do this
+by cleaning up pandas' internals and adding new methods do the extension array interface.
+
+String Dtype
+------------
+
+Currently, pandas stores text data in an ``object`` -dtype NumPy array.
+Each array stores Python strings. While pragmatic, since we rely on NumPy
+for storage and Python for string operations, this is memory inefficient
+and slow. We'd like to provide a native string type for pandas.
+
+The most obvious alternative is Apache Arrow. Currently, Arrow provides
+storage for string data. We can work with the Arrow developers to implement
+the operations needed for pandas users (for example, ``Series.str.upper``).
+These operations could be implemented in Numba (
+as prototyped in `Fletcher <https://github.com/xhochy/fletcher>`__)
+or in the Apache Arrow C++ library.
+
+Apache Arrow Interoperability
```

Comment:
```suggestion
Apache Arrow interoperability
```

(minor nit: we went with sentence case titles some time ago; some more below as well)